relay/server.go

```diff
@@ -226,14 +288,35 @@ func (s *Server) GetChunks(ctx context.Context, request *pb.GetChunksRequest) (*
 		bytesToSend = append(bytesToSend, bundleBytes)
 	}
 
-	return &pb.GetChunksReply{
-		Data: bytesToSend,
-	}, nil
+	return bytesToSend, nil
+}
+
+// computeChunkRequestRequiredBandwidth computes the bandwidth required to fulfill a GetChunks request.
+func computeChunkRequestRequiredBandwidth(request *pb.GetChunksRequest, mMap metadataMap) int {
+	requiredBandwidth := 0
+	for _, req := range request.ChunkRequests {
+		var metadata *blobMetadata
+		var requestedChunks int
+
+		if req.GetByIndex() != nil {
+			key := v2.BlobKey(req.GetByIndex().GetBlobKey())
+			metadata = mMap[key]
```

Comment:
check if the key exists in the map?

Reply:
Should not be possible, since process will fail before this point if there is missing metadata. But it's a useful safety check, so I don't mind adding it.